zebra-scan/src/tests.rs

```diff
@@ -0,0 +1,371 @@
+//! Test that we can scan the Zebra blockchain using the external `zcash_client_backend` crate
+//! scanning functionality.
+//!
+//! This tests belong to the proof of concept stage of the external wallet support functionality.
+
+use std::sync::Arc;
+
+use zcash_client_backend::{
+    proto::compact_formats::{
+        self as compact, ChainMetadata, CompactBlock, CompactSaplingOutput, CompactSaplingSpend,
+        CompactTx,
+    },
+    scanning::scan_block,
+};
+use zcash_note_encryption::Domain;
+use zcash_primitives::{
+    block::BlockHash,
+    consensus::{BlockHeight, Network},
+    constants::SPENDING_KEY_GENERATOR,
+    memo::MemoBytes,
+    sapling::{
+        note_encryption::{sapling_note_encryption, SaplingDomain},
+        util::generate_random_rseed,
+        value::NoteValue,
+        Note, Nullifier, SaplingIvk,
+    },
+    zip32::{AccountId, DiversifiableFullViewingKey, ExtendedSpendingKey},
+};
+
+use color_eyre::Result;
+
+use rand::{rngs::OsRng, RngCore};
+
+use ff::{Field, PrimeField};
+use group::GroupEncoding;
```

Comment:
Optional: our usual dependency groupings are: std, external dependencies, zebra dependencies, crate dependencies, module dependencies. But this is optional.

I like the split of `zcash_` and I think we should keep it.

```suggestion
use color_eyre::Result;
use ff::{Field, PrimeField};
use group::GroupEncoding;
use rand::{rngs::OsRng, RngCore};

use zcash_client_backend::{
    proto::compact_formats::{
        self as compact, ChainMetadata, CompactBlock, CompactSaplingOutput, CompactSaplingSpend,
        CompactTx,
    },
    scanning::scan_block,
};
use zcash_note_encryption::Domain;
use zcash_primitives::{
    block::BlockHash,
    consensus::{BlockHeight, Network},
    constants::SPENDING_KEY_GENERATOR,
    memo::MemoBytes,
    sapling::{
        note_encryption::{sapling_note_encryption, SaplingDomain},
        util::generate_random_rseed,
        value::NoteValue,
        Note, Nullifier, SaplingIvk,
    },
    zip32::{AccountId, DiversifiableFullViewingKey, ExtendedSpendingKey},
};

```

Reply:
This is going to cause a merge conflict, so let's not.